procdiscovery/pkg/inspectors/langdetect.go

```diff
@@ -24,25 +26,16 @@ var inspectorsList = []inspector{
 	&mysql.MySQLInspector{},
 }
 
-type LanguageDetectionResult struct {
-	Language common.ProgrammingLanguage
-}
+var ErrLanguageNotDetected = errors.New("language not detected")
 
-// DetectLanguage returns a list of all the detected languages in the process list
-func DetectLanguage(processes []process.Details) []LanguageDetectionResult {
-	var result []LanguageDetectionResult
-	for _, p := range processes {
-		for _, i := range inspectorsList {
-			language, detected := i.Inspect(&p)
-			if detected {
-				detectionResult := LanguageDetectionResult{
-					Language: language,
-				}
-				result = append(result, detectionResult)
-				break
-			}
+// DetectLanguage returns the detected language for the process
+func DetectLanguage(process process.Details) (common.ProgrammingLanguage, error) {
```

Comment:
It's not really an error if the language is not detected. I suggest returning something else like `found` or simply `common.UnknownProgrammingLanguage` to communicate that no language is detected but no err occured

Reply:
Changed to return `common.UnknownProgrammingLanguage`